operations/fetch.go

```diff
@@ -467,14 +467,24 @@ func fileNameWithIndex(filename string, index int) string {
 // truncateFilename truncates the filename (minus any extensions) so the entire filename length is less than the max
 func truncateFilename(fileName string) string {
 	if len(fileName) > fileNameMaxLength {
-		parts := strings.Split(fileName, ".")
-		if len(parts) == 0 {
-			return fileName
+		separatorIdx := strings.Index(fileName, ".tar")
+		// Check to see if there's a .tar extension somewhere in the file. Truncate the filename before ".tar".
+		// Otherwise, use "." as the separator. Truncate the filename before the final ".".
+		if separatorIdx == -1 {
+			separatorIdx = strings.LastIndex(fileName, ".")
+			if separatorIdx == -1 {
```

Comment:
I know this logic was wrong originally, but I believe we should still truncate fileName to be fileName[0:fileNameMaxLength], otherwise if the filename has no dots we just don't truncate anything.